### PR TITLE
refactor: Simplify favicon parsing

### PR DESCRIPTION
**Pull request by 12rambau**
reduce code complexity by tuning 2 methods of the extension. Normally I kept the functionality. It will be easier to manipulate for the automatic size PR.

## `create_favicons_meta`

Instead of 3 if statements with multiple calls to `generate_meta`, I changed a bit the logic:
- if the input is a `dict` transform it into a single element list
- if the item is a string (i.e. a single url) replace it with a single href favicon
- for each item in the list generate a tag and add it to a meta_favicons list
- return the meta_favicons joined by "\n"

## `_static_to_href`

Now the method is guessing automatically if href is absolute or not (no need for the `static-file` keyword anymore.
- transform `stati-file` into `href` (make sure legacy code continues to work)
- guess if the href is absolute (`/` is considered absolute)
- transform it if necessary directly in the favicon dict

### CODE REVIEW

File: sphinx_favicon/__init__.py [hunk 1]
```diff
@@ -113,31 +115,19 @@ def create_favicons_meta(pathto: Callable, favicons: FaviconsDef) -> Optional[st
...
-                attrs = {"href": favicon}
-            else:
-                attrs = _static_to_href(pathto, {FILE_FIELD: favicon})
-            meta_favicons += generate_meta(attrs) + "\n"
-    else:
-        logger.warning(
-            "Invalid config value for favicon extension."
-            "Custom favicons will notbe included in build."
-        )
-        return None
```
Comment (tcmetzger) [inline sphinx_favicon/__init__.py:137]: Why shouldn't we keep a warning like this?

File: sphinx_favicon/__init__.py [hunk 2]
```diff
@@ -91,12 +86,19 @@ def _static_to_href(pathto: Callable, favicon: Dict[str, str]) -> Dict[str, str]
...
-        attrs = favicon.copy()
-        attrs["href"] = pathto(
-            f"{OUTPUT_STATIC_DIR}/{attrs.pop(FILE_FIELD)}", resource=True
-        )
-        return attrs
+        favicon["href"] = favicon.pop(FILE_FIELD)
+
+    # check if link is absolute
+    link = favicon["href"]
+    is_absolute = bool(urlparse(link).netloc) or link.startswith("/")
```
Comment (tcmetzger) [inline sphinx_favicon/__init__.py:97]: This is great!

File: sphinx_favicon/__init__.py (hunk 1 above)
Comment (12rambau) [reply sphinx_favicon/__init__.py:137]: I though it was an overkill, but if you think people will set up things like this: 
```python
favicons = [
    ["https://secure.example.com/favicon/favicon-16x16.png"],
    ["https://secure.example.com/favicon/favicon-32x32.png"],
    ["https://secure.example.com/favicon/apple-touch-icon-180x180.png"],
]
```

Then ok... I set it back in the last commit !

Comment (tcmetzger) [reply sphinx_favicon/__init__.py:137]: Thanks! :)

**Comment by tcmetzger** [comment]
@12rambau Again, amazing work! :) 
I updated the docs to reflect the changes in this PR and also updated the static file test. 
Please check if these updates make sense, and add more tests / docs updates where necessary (e.g. for manifest / microsoft app settings?)

**Comment by 12rambau** [comment]
@tcmetzger thanks for the additions! 

My plan was to finish the 2 technical remaining PR (meta and auto-size) and once they are merged do a last round of verification for everything in a "prepare release" PR. I could have done everything in one but using iterative PR allows me to met requirements for: #16. 
Would you agree to merge this one (even if the test are not covering everything) so that I can finish the 2 others?

**Comment by tcmetzger** [comment]
@12rambau Absolutely, that sounds great!

**Review by tcmetzger — APPROVED** [review]
Thank you!!

**Pull request merged**